### DISCUSSION TITLE
I'm struggling to apply a couple simple tweaks, any help is truly appreciated!

### DISCUSSION THREAD

**Discussion by nerd190** [starship/starship#1462] [Q&A] [answered]
I only want to add two custom items. Where am I going wrong?
This is my starship.toml:

```bash
# ~/.config/starship.toml
# Disable the newline at the start of the prompt:
#add_newline = false
# Wait 10 milliseconds to check files in current directory:
#scan_timeout = 10

[custom.fish]
command = "echo 🐟"

[custom.username]
command = "echo getprop ro.product.device"

# Disable the package module, hiding it from the prompt completely:
[package]
disabled = true

[username]
disabled = true
 
# Overwrite a default_prompt_order and  use custom prompt_order:
prompt_order=["custom.fish","custom.username","directory","git_branch","git_commit","git_state","git_status","hg_branch", "golang","java","nodejs","php","python","ruby","rust","nix_shell","conda","memory_usage","env_var","cmd_duration","custom","line_break","jobs","battery","time","character"]

**Comment by matchai** [comment, marked as answer]
Hey @5c0tt-b0t 👋

Custom modules require a property specifying when the module should be shown. This can be specified with any of the following properties: `files`, `directories`, `extensions`, `when`.

If you always want these custom modules to be shown, the simplest thing to do would be to set `when` to a command that always returns a `0` status code:
```toml
[custom.fish]
command = "echo 🐟"
when = "true"
```

**Reply by nerd190** [reply]
WOW! **Thank you** for the quick response! Really is apprectiated!
Its looking NICE!
```
~/.config via 🐟 via OnePlus5T:
❯                                      18:04:21
```
My custom prompt_order is different though:
```
prompt_order=["custom.fish","custom.username","directory".....
```
So I was hoping for something like:
```
🐟OnePlus5T: ~/.config
❯                                      18:04:21
```
I was wondering if my ordering was wrong? Or perhaps I need to disable other items?

**Reply by matchai** [reply]
It looks like the `prompt_order` is not applying because it's not a root level property. Looking at the config snippet above, it is beneath the `[username]` heading. I'd suggest moving that line to be before any of the module headings.

**Reply by nerd190** [reply]
You are correct!! **Thank You!** 🤤 👇
```
🐟 via OnePlus5T in ~/.config
❯                                     18:40:16
```
So... To shut me up finally!
...I'm currently on a train setting this up from my phone! 😆  I like the verbosity of it on my PC (tons of columns to fill) but I would like to turn that off for my mobile device (less than 70 columns to fill!) so is it possible to:
-  Turn off: "on" / "in" / "via" etc.
- Get Fish's abbreviated parent directories back, e.g:
"~/s/e/0/Downloads"

Promise to stop bothering you after!  😇

**Reply by matchai** [reply]
Turning off the "on" / "in" / "via" separators will be possible once #346 is closed.

Fish truncation can be enabled via `fish_style_pwd_dir_length`:
```
[directory]
fish_style_pwd_dir_length = 1
```
